app/models/mark.rb

```diff
@@ -18,13 +18,29 @@ class Mark < ApplicationRecord
   validates_uniqueness_of :markable_id,
                           scope: [:result_id, :markable_type]
 
+  validates_inclusion_of :override, in: [true, false]
+
+  def calculate_deduction
+    return 0 if self.markable_type != 'FlexibleCriterion' || !self.override?
+    total_deduction = 0
+    applied_annotations = mark.result.annotations
+    applied_annotations.each do |annotation|
+      total_deduction += annotation.annotation_text.deduction ? annotation.annotation_text.deduction : 0
+    end
+    total_deduction
+  end
+
+  def update_deduction
+    self.mark = mark - calculate_deduction
+  end
+
   def scale_mark(curr_max_mark, prev_max_mark, update: true)
     return if mark.nil?
     return 0 if prev_max_mark == 0 || mark == 0 # no scaling occurs if prev_max_mark is 0 or mark is 0
     if markable.is_a? RubricCriterion
       new_mark = (mark * (curr_max_mark / prev_max_mark)).round(1)
     elsif markable.is_a? FlexibleCriterion
-      new_mark = (mark * (curr_max_mark.to_f / prev_max_mark)).round(2)
+      new_mark = (mark * (curr_max_mark.to_f / prev_max_mark)).round(2) - :calculate_deduction
```

Comment:
Let's not do it like this. This method is called when the flexible criterion's `max_mark` changes. Look for that callback, and then scale the criterion's annotation deductions there as well.